repo2docker/detectors.py

```diff
@@ -37,14 +38,56 @@ def detect(self, workdir):
         return os.path.exists(os.path.join(workdir, 'Dockerfile'))
 
     def build(self, workdir, ref, output_image_spec):
-        client = docker.APIClient(base_url='unix://var/run/docker.sock', version='auto')
+        client = docker.APIClient(version='auto', **kwargs_from_env())
         for progress in client.build(
                 path=workdir,
                 tag=output_image_spec,
                 decode=True
         ):
             if 'stream' in progress:
-                self.log.info(progress['stream'], extra=dict(phase='building'))
+                if self.capture:
+                    self.log.info(progress['stream'], extra=dict(phase='building'))
+                else:
+                    sys.stdout.write(progress['stream'])
+
+_legacy_binder_docker_appendix = """
```

Comment:
can you move this to be a traitlet under the Legacy Builder? keeps it contained and out of the global namespace of the file, + possibly allows overrides if needed...